parse.d: Divide expression into two ifs

wilzbach opened the pull request.

Geod24 committed:
Remove a deprecated CommaExp usage

Geod24 committed:
Remove usage of implicit string concatenation

Geod24 committed:
test/runnable/sieve.d: Use D-style arrays

LemonBoy committed:
Fix issue #16278 - Do not emit TypeInfo for speculative instantiations.

LemonBoy committed:
Fix issue 17073 - Do not ignore the explicit initializers

We don't really care about the _init being void-initialized when the
user has supplied an explicit value for a given field of the aggregate.

WalterBright committed:
fix covariant behavior with 'return scope'

LemonBoy committed:
Fix issue 16598 - Explicitly discard the exp. value in void CondExp

WalterBright committed:
refactor: split off StatementRewriteWalter to statementrww.d

MartinNowak committed:
Merge pull request #6433 from WalterBright/cod4-CodeBuilder-9

cod4.c: use CodeBuilder part 9 cdshtlng()

dlang-bot committed:
Merge pull request #6427 from LemonBoy/b17073

Fix issue 17073 - Do not ignore the explicit initializers
merged-on-behalf-of: Martin Nowak <code@dawg.eu>

mathias-lang-sociomantic committed:
Fix issue 17086: DMD segfault with multiple template matches and invalid code

The two conditions in mtype and expression can be simplified, because in the event of 'e' being , we end up with a segfault.
Moreover, the code was just missing a couple of null checks.

LemonBoy committed:
Fix issue 6400 - Better interaction between with() and opDispatch

UplinkCoder committed:
use dyncast.object; add dyncast.statement - change return type of dyncast so final switch will work

WalterBright committed:
Merge pull request #6438 from mathias-lang-sociomantic/segv

Fix issue 17086: DMD segfault with multiple template matches and invalid code

Commit:
fix issue 16245 - the message emitted when a const function mutates members is misleading

WalterBright committed:
cod4.c: use CodeBuilder part 10 cdshtlng()

MartinNowak committed:
Merge pull request #6443 from WalterBright/cod4-CodeBuilder-10

cod4.c: use CodeBuilder part 10 cdshtlng()

MartinNowak committed:
Merge pull request #6434 from LemonBoy/b16598

Fix issue 16598 - Explicitly discard the exp. value in void CondExp

rainers committed:
Fix issue 16691: make value parameters implicitely passed by pointer on Win64 r-value references

WalterBright committed:
cod4.c: use CodeBuilder part 11 cdbyteint()

WalterBright committed:
Merge pull request #6441 from BBasile/issue-16245

fix issue 16245 - the message emitted when a const function mutates members is misleading

WalterBright committed:
Merge pull request #6442 from UplinkCoder/fixDyncast

Fix dyncast

WalterBright committed:
cod4.c: use CodeBuilder part 12

MartinNowak committed:
Merge remote-tracking branch 'upstream/stable' into merge_stable

MartinNowak committed:
Merge pull request #6448 from MartinNowak/merge_stable

Merge remote-tracking branch 'upstream/stable' into merge_stable

WalterBright committed:
cod4.c: use CodeBuilder part 13

WalterBright committed:
cod4.c: use CodeBuilder part 14

WalterBright committed:
Merge pull request #6429 from WalterBright/scope-covariant

fix covariant behavior with 'return scope'

WalterBright committed:
Merge pull request #6439 from LemonBoy/b6400

Fix issue 6400 - Better interaction between with() and opDispatch

MartinNowak committed:
Merge pull request #6450 from WalterBright/cod4-CodeBuilder-14

cod4.c: use CodeBuilder part 14

MartinNowak committed:
Merge pull request #6449 from WalterBright/cod4-CodeBuilder-13

cod4.c: use CodeBuilder part 13

MartinNowak committed:
Merge pull request #6447 from WalterBright/cod4-CodeBuilder-12

cod4.c: use CodeBuilder part 12

MartinNowak committed:
Merge pull request #6446 from WalterBright/cod4-CodeBuilder-11

cod4.c: use CodeBuilder part 11 cdbyteint()

LemonBoy committed:
Fix issue 16483 - Prevent an ICE with lambdas as template parameter

WalterBright committed:
fix Issue 17072 - [REG 2.073.0-b1] missing symbols with -inline

WalterBright committed:
cod2.c: use CodeBuilder part 4 cdmul()

UplinkCoder committed:
make dyncast const - and move DYNCAST out of globals into root

yebblies committed:
Merge pull request #6455 from UplinkCoder/const_dyncast

make dyncast const - and move DYNCAST out of globals into root

WalterBright committed:
Merge pull request #6445 from rainers/issue16691

Fix issue 16691: win64 debug info: pass struct value parameters by reference, not pointer

LemonBoy committed:
Fix issue 16697 - Accept __vector as type specialization in is()

MartinNowak committed:
permute args with -dip25

MartinNowak committed:
don't acutally sleep in test

MartinNowak committed:
Merge pull request #6452 from WalterBright/fix17072

fix Issue 17072 - [REG 2.073.0-b1] missing symbols with -inline

WalterBright committed:
fix Issue 17087 - [REG2.072] Wrong generated with cfloat and creal when casting from int

WalterBright committed:
cod2.c: use CodeBuilder part 5 cdnot()

LemonBoy committed:
Fix issue 15428 - Properly detach the temporary scope for compiles()

LemonBoy committed:
Fix issue 15676 - Don't omit the attributes for the postblit constructor

WalterBright committed:
cod2.c: use CodeBuilder part 6 cdcom() cdbswap()

MartinNowak committed:
Merge pull request #6460 from WalterBright/fix17087

fix Issue 17087 - [REG2.072] Wrong generated with cfloat and creal wh…

MartinNowak committed:
Merge pull request #6454 from WalterBright/cod2-CodeBuilder-4

cod2.c: use CodeBuilder part 4 cdmul()

MartinNowak committed:
Merge pull request #6461 from WalterBright/cod2-CodeBuilder-5

cod2.c: use CodeBuilder part 5 cdnot()

MartinNowak committed:
Merge pull request #6351 from LemonBoy/b16278

Fix issue #16278 - Do not emit TypeInfo for speculative instantiations.

MartinNowak committed:
Merge pull request #6465 from WalterBright/cod2-CodeBuilder-6

cod2.c: use CodeBuilder part 6 cdcom() cdbswap()

WalterBright committed:
cod2.c: use CodeBuilder part 7 cdcond()

rainers committed:
add changelog entry for -mscrt=lib

WalterBright committed:
self-assignment should still allow 'scope' inference

MartinNowak committed:
Merge pull request #6467 from rainers/mscrt_changelog

changelog entry for -mscrt=lib

MartinNowak committed:
Merge remote-tracking branch 'upstream/stable' into merge_stable

dlang-bot committed:
Merge pull request #6470 from MartinNowak/merge_stable

Merge remote-tracking branch 'upstream/stable' into merge_stable
merged-on-behalf-of: Martin Nowak <code@dawg.eu>

rainers committed:
Update mscrtlib-option.dd

WalterBright committed:
cod2.c: use CodeBuilder part 8 cdloglog()

WalterBright committed:
Merge pull request #6458 from LemonBoy/b16697

Fix issue 16697 - Accept __vector as type specialization in is()

MartinNowak committed:
Merge pull request #6471 from dlang/rainers-patch-2

fix typo in mscrtlib-option.dd

jacob-carlborg committed:
Add missing changelog entry for new default Ddoc theme

UplinkCoder committed:
Add astTypeName-visitor to print out the name of a ast-node

LemonBoy committed:
Fix issue 17111 - More robust detection of variables used as case exp.

WalterBright committed:
infer 'scope' for 'this' parameter

andralex committed:
Merge pull request #6478 from WalterBright/infer-scope-this

infer 'scope' for 'this' parameter

MartinNowak committed:
details and wording for ddoc changelog entry

MartinNowak committed:
Merge pull request #6475 from jacob-carlborg/changelog_new_ddoc_theme

Add missing changelog entry for new default Ddoc theme

MartinNowak committed:
purge changelog

MartinNowak committed:
Merge remote-tracking branch 'upstream/stable' into merge_stable

MartinNowak committed:
update VERSION

MartinNowak committed:
Merge pull request #6466 from WalterBright/cod2-CodeBuilder-7

cod2.c: use CodeBuilder part 7 cdcond()

MartinNowak committed:
Merge pull request #6472 from WalterBright/cod2-CodeBuilder-8

cod2.c: use CodeBuilder part 8 cdloglog()

ibuclaw committed:
More synchronizing C++ headers with D code.

dlang-bot committed:
Merge pull request #6481 from MartinNowak/version

update VERSION
merged-on-behalf-of: Martin Nowak <code@dawg.eu>

MartinNowak committed:
Merge pull request #6131 from UplinkCoder/asttypename

add astTypeName to help with AST-exploration

WalterBright committed:
Merge branch 'master' into self-assign

WalterBright committed:
[scope] escaping through ref parameter

andralex committed:
Merge pull request #6469 from WalterBright/self-assign

self-assignment should still allow 'scope' inference

WalterBright committed:
cod2.c: use CodeBuilder part 9 cdshift()

WalterBright committed:
fix Issue 17117 - erroneous 'escaping reference to local variable'

andralex committed:
Merge pull request #6451 from LemonBoy/b16483

Fix issue 16483 - Prevent an ICE with lambdas as template parameter

Commit:
Fix crash in ld_mod.

WalterBright committed:
scope: parameter does not escape through itself

rainers committed:
Merge pull request #6492 from sprinkle131313/master-ldfpu

Fix crash in ld_mod.

MartinNowak committed:
Merge pull request #6487 from WalterBright/cod2-CodeBuilder-9

cod2.c: use CodeBuilder part 9 cdshift()

MartinNowak committed:
Merge pull request #6488 from WalterBright/fix17117

fix Issue 17117 - erroneous 'escaping reference to local variable'

MartinNowak committed:
Merge pull request #6483 from WalterBright/ref-scope-escape

[scope] escaping through ref parameter

WalterBright committed:
fix Issue 17123 - [REG 2.073] Issues with return @safe inference

WalterBright committed:
[scope] fix ambiguity in scope mangling

WalterBright committed:
Merge branch 'master' into scope-self-ref

WalterBright committed:
fix Issue 17123 - [REG 2.073] Issues with return @safe inference

MartinNowak committed:
Merge remote-tracking branch 'upstream/stable' into merge_stable

MartinNowak committed:
Merge pull request #6497 from WalterBright/fix1707-stable

fix Issue 17123 - [REG 2.073] Issues with return @safe inference

MartinNowak committed:
Merge pull request #6480 from MartinNowak/merge_stable

Merge remote-tracking branch 'upstream/stable' into merge_stable

Ingrater committed:
Add missing root/hash.d to VisualD project.

WalterBright committed:
cod2.c: use CodeBuilder part 10 cdind()

WalterBright committed:
reboot #6081: Fix issue 14859 - 16MiB size limit for static array

WalterBright committed:
cod2.c: use CodeBuilder part 11 cdstrlen()

RazvanN7 committed:
Fix Issue 17057 - Added test file

andralex committed:
Merge pull request #6505 from RazvanN7/Issue_17057

Fix Issue 17057 - trait allMembers incorrectly includes imports

rainers committed:
Merge pull request #6500 from Ingrater/fixVSBuild

Fix Visual Studio Solution of dmd.

WalterBright committed:
cod2.c: use CodeBuilder part 12 cdmemcmp()

WalterBright committed:
Merge pull request #6498 from MartinNowak/merge_stable

Merge remote-tracking branch 'upstream/stable' into merge_stable

WalterBright committed:
cod1.c: CodeBuilder part 3 movParams()

MartinNowak committed:
Merge pull request #6502 from WalterBright/cod2-CodeBuilder-10

cod2.c: use CodeBuilder part 10 cdind()

MartinNowak committed:
Merge pull request #6504 from WalterBright/cod2-CodeBuilder-11

cod2.c: use CodeBuilder part 11 cdstrlen()

MartinNowak committed:
Merge pull request #6509 from WalterBright/cod1-CodeBuilder-3

cod1.c: CodeBuilder part 3 movParams()

MartinNowak committed:
Merge pull request #6508 from WalterBright/cod2-CodeBuilder-12

cod2.c: use CodeBuilder part 12 cdmemcmp()

WalterBright committed:
cod1.c: CodeBuilder part 4 tstresult()

rainers committed:
Merge pull request #6496 from WalterBright/scope-mangle

[scope] fix ambiguity in scope mangling

MaskRay committed:
fix SIGFPE caused by integer division overflow `{int,long}.min / (-1)` in constant folding

WalterBright committed:
cod1.c: CodeBuilder part 5 fixresult()

MartinNowak committed:
fix Issue 17130 - ambiguous implicit super call

- need type of this to resolve base class ctor overloads

WalterBright committed:
cod1.c: CodeBuilder part 6 callclib()

WalterBright committed:
[scope] infer 'return scope'

epi committed:
Fix Issue 17143 - [REG2.072.0] Declaration is already defined on global enum = tuple(...).expand declaration

Regression was introduced in 53a70e4d66b8bf1b96a51c06eeb2df7fa836f647

MartinNowak committed:
Merge pull request #6512 from WalterBright/cod1-CodeBuilder-5

cod1.c: CodeBuilder part 5 fixresult()

MartinNowak committed:
Merge pull request #6515 from WalterBright/cod1-CodeBuilder-6

cod1.c: CodeBuilder part 6 callclib()

MartinNowak committed:
Merge pull request #6510 from WalterBright/cod1-CodeBuilder-4

cod1.c: CodeBuilder part 4 tstresult()

MartinNowak committed:
retest

ibuclaw committed:
Declarations inside a C++ namespace should also be marked isdataseg

MartinNowak committed:
Merge pull request #6517 from epi/fix-17143

Fix Issue 17143 - [REG2.072.0] Declaration is already defined on global enum = tuple(...).expand declaration

WalterBright committed:
cod1.c: CodeBuilder part 7 funccall()

WalterBright committed:
cod1.c: CodeBuilder part 8 getlvalue()

WalterBright committed:
cod1.c: CodeBuilder part 9 funccall()

WalterBright committed:
cgcod.c: CodeBuilder pt 1 prolog()

WalterBright committed:
cgcod.c: CodeBuilder pt 2

MartinNowak committed:
Merge pull request #6522 from WalterBright/cgcod-CodeBuilder-1

cgcod.c: CodeBuilder pt 1 prolog()

MartinNowak committed:
Merge pull request #6523 from WalterBright/cgcod-CodeBuilder-2

cgcod.c: CodeBuilder pt 2

WalterBright committed:
[scope] propagate 'scope' to tuple elements

WalterBright committed:
Merge pull request #6518 from ibuclaw/nsdataseg

Declarations inside a C++ namespace should also be marked isdataseg

MartinNowak committed:
update version

e-y-e committed:
Add RISCV32 and RISCV64 version identifiers

MartinNowak committed:
Merge remote-tracking branch 'upstream/stable' into merge_stable

# Conflicts:
#	VERSION

WalterBright committed:
[scope] recognize 'return' in scanning types

UplinkCoder committed:
fix Issue 17168

andralex committed:
Merge pull request #6528 from WalterBright/parameter-return

[scope] recognize 'return' in scanning types

WalterBright committed:
remove helper functions for assert and unittest

MartinNowak committed:
Merge pull request #6513 from MartinNowak/fix17130

fix Issue 17130 - ambiguous implicit super call

andralex committed:
Merge pull request #6516 from WalterBright/infer-return-scope

[scope] infer 'return scope'

andralex committed:
Merge pull request #6503 from WalterBright/fix14859

reboot #6081: Fix issue 14859 - 16MiB size limit for static array

andralex committed:
Merge pull request #6519 from WalterBright/cod1-CodeBuilder-7

cod1.c: CodeBuilder part 7 funccall()

andralex committed:
Merge pull request #6520 from WalterBright/cod1-CodeBuilder-8

cod1.c: CodeBuilder part 8 getlvalue()

andralex committed:
Merge pull request #6521 from WalterBright/cod1-CodeBuilder-9

cod1.c: CodeBuilder part 9 funccall()

andralex committed:
Merge pull request #6493 from WalterBright/scope-self-ref

scope: parameter does not escape through itself

MartinNowak committed:
remove phobos from test-case

MartinNowak committed:
Merge pull request #6477 from LemonBoy/b17111

Fix issue 17111 - More robust detection of variables used as case exp.

MartinNowak committed:
Merge pull request #6463 from LemonBoy/b15676

Fix issue 15676 - Don't omit the attributes for the postblit constructor

MartinNowak committed:
add test case from bug report

MartinNowak committed:
Merge pull request #6462 from LemonBoy/b15428

Fix issue 15428 - Properly detach the temporary scope for compiles()

MartinNowak committed:
Merge pull request #6524 from WalterBright/scope-tuple

[scope] propagate 'scope' to parameter tuple elements

MartinNowak committed:
Merge pull request #6527 from MartinNowak/merge_stable

Merge remote-tracking branch 'upstream/stable' into merge_stable

MartinNowak committed:
add test case for Issue 4682

MartinNowak committed:
fix % in printf formatted error message

MartinNowak committed:
Merge pull request #6529 from UplinkCoder/fix_elshl

fix Issue 17168 - refactor elshl

MartinNowak committed:
Merge pull request #6511 from MaskRay/master

fix SIGFPE caused by integer division overflow `sinteger_t.min / (-1)` in constant folding

rainers committed:
Win32: write references to mutable pointers in data and tls segment

rainers committed:
add test case for pointer references

rainers committed:
merge appropriate druntime PR

andralex committed:
Merge pull request #6530 from WalterBright/rm-assert-unittest

remove helper functions for assert and unittest

andralex committed:
Merge pull request #6495 from WalterBright/fix17123

[scope] improve 'scope' and 'return' inference for delegates

Abscissa committed:
Fix bugzilla 14413: Spurious newline in ddoc JSON output for multiple successive line doc comments

CyberShadow committed:
Merge pull request #4745 from Abscissa/issue-14413

Fix bugzilla 14413: Spurious newline in ddoc JSON output for multiple…

rainers committed:
add function documentation

WalterBright committed:
eliminate array bounds error helper function

rainers committed:
implement Obj::write_pointerRef for MS-COFF

andralex committed:
Merge pull request #6436 from WalterBright/add-statementrww.d

refactor: split off StatementRewriteWalker to statementrww.d

jacob-carlborg committed:
Fix issue 17169 - New default ddoc theme merges all paragraphs except the first

MartinNowak committed:
fix Issue 17130 - ambiguous implicit super call

- need to pass class type to resolveFuncCall
- also make implicit ctor shared for shared base ctor

rainers committed:
fix travis builds: strtol needs a common signature for dmd 2.073 and later

rainers committed:
workaround chanaged signature of qsort

WalterBright committed:
Merge pull request #6539 from rainers/fix_travis

fix travis builds: strtol needs a common signature for dmd 2.073 and later

WalterBright committed:
put string literals in COMDAT sections

MartinNowak committed:
Merge pull request #6541 from MartinNowak/fix17130

fix Issue 17130 - ambiguous implicit super call

MartinNowak committed:
Merge pull request #6538 from jacob-carlborg/issue_17169

Fix issue 17169 - New default ddoc theme merges all paragraphs except…

MartinNowak committed:
Merge remote-tracking branch 'upstream/stable' into merge_stable

dlang-bot committed:
Merge pull request #6537 from MartinNowak/merge_stable

Merge remote-tracking branch 'upstream/stable' into merge_stable
merged-on-behalf-of: Martin Nowak <code@dawg.eu>

andralex committed:
Merge pull request #6542 from WalterBright/string-comdat

put string literals in COMDAT sections

andralex committed:
Merge pull request #6536 from WalterBright/darrayp

eliminate array bounds error helper function

WalterBright committed:
put string literals in readonly comdats

WalterBright committed:
add backend/md5.d

andralex committed:
Merge pull request #6544 from WalterBright/readonly-comdat

put string literals in readonly comdats

andralex committed:
Merge pull request #6545 from WalterBright/add-md5.d

add backend/md5.d

WalterBright committed:
use md5 to generate names for string comdats

aG0aep6G committed:
TypeInfo.init -> .initializer

TypeInfo.init is deprecated and is going to to be removed.

dlang-bot committed:
Merge pull request #6547 from aG0aep6G/TypeInfo.init

TypeInfo.init -> .initializer
merged-on-behalf-of: David Nadlinger <code@klickverbot.at>

WalterBright committed:
Merge pull request #6526 from e-y-e/riscv-version

Add RISCV32 and RISCV64 version identifiers

WalterBright committed:
write Elf string literals to .rodata.strM.N sections

rainers committed:
updated comments to use DDoc style

rainers committed:
OMF: move pointer reference data to CODE segment

WalterBright committed:
Merge pull request #6534 from rainers/data_ptrref

[WIP] write references to mutable pointers in data and tls segment

rainers committed:
fixup merge of PR6534

MartinNowak committed:
Merge pull request #6549 from rainers/fix_makefile

fixup merge of PR6534

kinke committed:
Add missing field in C++ global.params struct

WalterBright committed:
Merge pull request #6546 from WalterBright/md5-comdat

use md5 to generate names for string comdats

WalterBright committed:
Merge pull request #6551 from kinke/stable

[Trivial] Add missing field in C++ global.params struct

WalterBright committed:
Merge pull request #6548 from WalterBright/rodata.str

write Elf string literals to .rodata.strM.N sections

WalterBright committed:
machobj.c: use __cstring section for string literals

wilzbach committed:
posix.mak: Add style target to simplify CI setup

wilzbach committed:
parse.d: Divide expression into two ifs

WalterBright committed:
refactor: split off dw_writeToObj() in backend/out.c

andralex committed:
Merge pull request #6553 from WalterBright/__cstring

machobj.c: use __cstring section for string literals

dlang-bot committed:
Merge pull request #6557 from WalterBright/dt_writeObj

refactor: split off dw_writeToObj() in backend/out.c
merged-on-behalf-of: Martin Nowak <code@dawg.eu>

UplinkCoder committed:
Add Dyncast.condition

dlang-bot committed:
Merge pull request #6555 from wilzbach/posix-add-style-target

posix.mak: Add style target to simplify CI setup
merged-on-behalf-of: Martin Nowak <code@dawg.eu>

WalterBright committed:
Windows: put ModuleInfo in read-only section

RazvanN7 committed:
Added  to generated path

UplinkCoder committed:
remove c++ style DYNCAST_*

rainers committed:
Merge pull request #6559 from WalterBright/minfo-readonly

Windows: put ModuleInfo in read-only section

WalterBright committed:
Merge pull request #6563 from UplinkCoder/rewrite_dyncast

remove c++ style DYNCAST_*

MartinNowak committed:
fix Issue 17215 -  ICE(cgcod.c:findreg) with SIMD and -O -inline

- caused by assigning 2*REGSIZE XMM vectors partially
- only triggers since SROA sees this 8 bytes access

WalterBright committed:
add SHF_MERGE and SHF_STRINGS to .rodata.strM.N

MartinNowak committed:
Merge pull request #6558 from UplinkCoder/dyncast_condition

Add Dyncast.condition

MartinNowak committed:
Merge pull request #6567 from WalterBright/string-merge

elfobj.c: add SHF_MERGE and SHF_STRINGS to .rodata.strM.N

WalterBright committed:
Merge pull request #6566 from MartinNowak/fix17215

fix Issue 17215 -  ICE(cgcod.c:findreg) with SIMD and -O -inline

UplinkCoder committed:
avoid introducing irritating indentation on  if {} else if {} constructs

UplinkCoder commented on src/parse.d:
```diff
@@ -2866,24 +2866,29 @@ final class Parser : Lexer
+                                ai = token.ident;
+                                nextToken();
+                            }
+                            else goto _else;
```
no need for the else branch.

MartinNowak committed:
Merge remote-tracking branch 'upstream/stable' into merge_stable

MartinNowak committed:
Merge pull request #6569 from MartinNowak/merge_stable

Merge remote-tracking branch 'upstream/stable' into merge_stable

UplinkCoder committed:
add test

Commit:
issue 6760 - deprecate `@disable` on overridden and deprecated functions

WalterBright committed:
Merge pull request #6570 from BBasile/issue-6760

issue 6760 - deprecate `@disable` on overridden and deprecated functions

ibuclaw committed:
Remove repeated call to new VarExp

WalterBright committed:
Merge pull request #6575 from ibuclaw/more-sync-headers

More synchronizing C++ headers with D code.

RazvanN7 committed:
Uncommented previously commented if at line 522

CyberShadow committed:
posix.mak: Use TMP environment variable for temporary directory

Allows working around issues such as /tmp filesystems mounted noexec,
or conflicts due to multiple users building D on the same machine.

dlang-bot committed:
Merge pull request #6562 from RazvanN7/ddmd_generated

Generate all .o .deps and binaries to generated/
merged-on-behalf-of: unknown

yebblies committed:
Suppress duplicate errors on integer division overflow during constfolding

dlang-bot committed:
Merge pull request #6579 from yebblies/overflowdup

Suppress duplicate errors on integer division overflow during constfolding
merged-on-behalf-of: Andrei Alexandrescu <andralex@users.noreply.github.com>

dlang-bot committed:
Merge pull request #6574 from ibuclaw/double-new

Remove repeated call to new VarExp
merged-on-behalf-of: David Nadlinger <code@klickverbot.at>

andralex committed:
Merge pull request #6568 from UplinkCoder/patch_hdrgen

HdrGen: avoid introducing irritating indentation on  if {} else if {} constructs

dlang-bot committed:
Merge pull request #6578 from CyberShadow/pull-20170227-153912

posix.mak: Use TMP environment variable for temporary directory
merged-on-behalf-of: Sebastian Wilzbach <sebi.wilzbach@gmail.com>

RazvanN7 committed:
Fix Issue 15616 - missing candidate in error message (#6525)

Fix Issue 15616 - missing candidate in error message
merged-on-behalf-of: Andrei Alexandrescu <andralex@users.noreply.github.com>

wilzbach committed:
Use Phobos doc generation (#6581)

DDoc doc generation: use similar Makefile target as Phobos (whole paths with underscores)
merged-on-behalf-of: Andrei Alexandrescu <andralex@users.noreply.github.com>

ibuclaw committed:
fix Issue 17237 - Wrong alignment for 256-bit vectors.

WalterBright committed:
Merge pull request #6582 from ibuclaw/issue17237

fix Issue 17237 - Wrong alignment for 256-bit vectors.

wilzbach committed:
osmodel.mak: Mention tools + dlang.org repos in the update reminder

WalterBright committed:
Merge pull request #6583 from wilzbach/os-model-mention-tools-and-dlang-org

osmodel.mak: Mention tools + dlang.org repos in the update reminder

RazvanN7 committed:
Moved everything that is generated to the generated folder (#6580)

Moved everything that is generated to the generated folder
merged-on-behalf-of: Andrei Alexandrescu <andralex@users.noreply.github.com>

wilzbach committed:
Fix dependency chain in posix.mak

MartinNowak committed:
fix calling win makefiles w/ MODEL=32 (#6585)

fix calling win makefiles w/ MODEL=32
merged-on-behalf-of: Andrei Alexandrescu <andralex@users.noreply.github.com>

wilzbach committed:
Automatically generate dmd.conf (once)

MartinNowak committed:
fixes make -f posix.mak html AUTO_BOOTSTRAP=1

- tries to builds HTMLS w/o having downloaded the bootstrap compiler
- current state breaks builds script for dlang.org

MartinNowak committed:
update VERSION

MartinNowak committed:
Merge remote-tracking branch 'upstream/stable' into merge_stable

# Conflicts:
#	VERSION

dlang-bot committed:
Merge pull request #6587 from MartinNowak/fix_html_target

fixes make -f posix.mak html AUTO_BOOTSTRAP=1
merged-on-behalf-of: Martin Nowak <code@dawg.eu>

dlang-bot committed:
Merge pull request #6586 from wilzbach/makefile-fix-dependency-chain

Fix dependency chains in posix.mak
merged-on-behalf-of: Martin Nowak <code@dawg.eu>

MartinNowak committed:
Merge pull request #6588 from MartinNowak/merge_stable

Merge remote-tracking branch 'upstream/stable' into merge_stable

Geod24 commented:
Thanks for taking care of this :)
